discovery_engine_core/web-api/src/storage.rs

```diff
@@ -107,6 +107,15 @@ impl UserState {
     {
         let mut tx = self.pool.begin().await?;
 
+        sqlx::query("INSERT INTO coi_update_lock (user_id) VALUES ($1) ON CONFLICT DO NOTHING;")
+            .bind(user_id.as_ref())
+            .execute(&mut tx)
+            .await?;
+        sqlx::query("SELECT FROM coi_update_lock WHERE user_id = $1 FOR UPDATE;")
+            .bind(user_id.as_ref())
+            .execute(&mut tx)
+            .await?;
+
```

Comment:
not sure if we still need the old lock in line 125, below 👇 

```rust
let mut positive_cois: Vec<_> = sqlx::query_as::<_, QueriedCoi>(
  "SELECT coi_id, is_positive, embedding, view_count, view_time_ms, last_view 
    FROM center_of_interest 
    WHERE user_id = $1 AND is_positive 
    FOR UPDATE;", // do we need this line ???
)
```

Reply:
shouldn't be necessary